lib/manager/components/deployment/PeliasPanel.js

```diff
@@ -0,0 +1,213 @@
+    const { target } = event
+    const { value, id } = target
+    this._updateDeployment({ [id]: value })
+  };
+
+  /**
+   * Takes files submitted via the file uploader modal and sends them to datatools server where they are
+   * uploaded to S3
+   * @param files Array of files returned by the file upload modal
+   * @returns     True or false depending on if the upload was a success
+   */
+  _onConfirmUpload = async (files: Array<File>) => {
+    const { deployment, uploadPeliasWebhookCsvFile } = this.props
+    const { fileToDeleteOnSuccesfulUpload } = this.state
+
+    const file = files[0]
+    // TODO: more extensive csv validation?
+    if (file.type !== 'text/csv') {
+      return false
+    }
+    // Without the redundant await, flow is unhappy
+    // eslint-disable-next-line no-return-await
+    return await uploadPeliasWebhookCsvFile(deployment, file, fileToDeleteOnSuccesfulUpload)
+  }
+
+  /**
+   * Removes a csv file from the list of Pelias csv files associated with a deployment
+   * WARNING: DOES NOT REMOVE THE FILE FROM S3!
+   * @param url   The URL of the csv file to remove from the deployment
+   */
+  _deleteCsvFile = async (url: string) => {
+    const { deployment } = this.props
+    if (!deployment.peliasCsvFiles) return
+
+    const updatedCsvFiles = deployment.peliasCsvFiles.filter(u => u !== url)
+    this._updateDeployment({'peliasCsvFiles': updatedCsvFiles})
+  }
+
+  /**
+   * Persists changes to the deployment object by making a request to the datatools server
+   */
+  _updateDeployment = (props: { [string]: any }) => {
+    const { deployment, updateDeployment } = this.props
+    updateDeployment(deployment, props)
+  };
+
+  /**
+   * Renders a csv link associated with the deployment. Renders the file name (or URI if not added by datatools) and
+   * buttons to replace or remove the file
+   * @param {*} url     The URL to add to the list of csv files associated with the deployment
+   * @param {*} enabled Whether the buttons should be enabled
+   * @returns           JSX including the file name and buttons
+   */
+  renderCsvUrl = (url: string, enabled: boolean) => {
+    // Usually, files will be rendered by https://github.com/ibi-group/datatools-server/blob/dev/src/main/java/com/conveyal/datatools/manager/controllers/api/DeploymentController.java
+    // so we can take advantage of a predictable filename
+    // As a fallback, render the full url
+    const fileName = url.split('_').length === 2 ? url.split('_')[1] : url
+    return (
+      <li key={url}>
+        {fileName}{' '}
+        <Button disabled={!enabled} bsSize='xsmall' onClick={() => {
+          this.refs.uploadModal.open()
+          this.setState({fileToDeleteOnSuccesfulUpload: url})
+        }}>
+          Replace
+        </Button>
+        <Button disabled={!enabled} bsSize='xsmall' onClick={() => {
+          if (confirm('Are you sure you want to delete this CSV file?')) {
+            this._deleteCsvFile(url)
+          }
```

Comment:
Can you move these onClick handlers to be in class methods?

Reply:
Yes they can! In c3e8b6e